modules/core_project_factory/main.tf

```diff
@@ -310,6 +310,13 @@ resource "google_storage_bucket" "project_bucket" {
   uniform_bucket_level_access = var.bucket_ula
   public_access_prevention    = var.bucket_pap
 
+  dynamic "soft_delete_policy" {
+    for_each = var.soft_delete_policy == {} ? [] : [var.soft_delete_policy]
```

Comment:
Suggestion only: might be clearer to have the "enabled" outcome first, but checking `!= {}`